code/modules/shuttle/supply.dm

```diff
@@ -187,6 +188,30 @@
 					SSeconomy.research_designs += design.id
 					msg += "<span class='good'>+[SSeconomy.credits_per_design]</span>: [design.name] design.<br>"
 
+				// Sell viral sample virology goals
+				if(istype(thing, /obj/item/reagent_containers))
+					var/obj/item/reagent_containers/C = thing
+					if(C.reagents && C.reagents.reagent_list)
+						for(var/datum/virology_goal/G in GLOB.virology_goals)
+							if(G.completed)
+								continue
+							if(G.check_completion(C.reagents.reagent_list))
+								medical_credits += SSeconomy.credits_per_virology_goal
+								msg += "<span class='good'>+[SSeconomy.credits_per_virology_goal]</span>: [G.name] completion.<br>"
+
+				// Sell viral sample virology goals that are in vial lockboxes too
+				if(istype(thing, /obj/item/storage/lockbox/vials))
+					for(var/thing2 in thing)
```

Comment:
rename thing2 to `thing_content` or something.
Never name variables to just satisfy the compiler. Remember, that other coders will also use your code :D